apps/src/convert.cpp

```diff
@@ -1,83 +1,72 @@
 #include <iostream>
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
 #include <opencv2/core.hpp>
 #include <opencv2/imgcodecs.hpp>
 #include <opencv2/photo.hpp>
 
-#include <boost/filesystem.hpp>
+#include "envitools/EnviUtils.hpp"
 
 namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+namespace et = envitools;
 
 constexpr static uint16_t MAX_INTENSITY = std::numeric_limits<uint16_t>::max();
 
-// return the filenames of all files that have the specified extension
-// in the specified directory and all subdirectories
-std::vector<fs::path> FindByExtension(
-    const fs::path root, const std::string ext)
-{
-    std::vector<fs::path> ret;
-    if (!fs::exists(root) || !fs::is_directory(root)) {
-        return ret;
-    }
-
-    fs::recursive_directory_iterator it(root);
-    fs::recursive_directory_iterator endit;
-
-    while (it != endit) {
-        if (fs::is_regular_file(*it) && it->path().extension() == ext) {
-            fs::path path = canonical(it->path());
-            ret.push_back(path);
-        }
-        ++it;
-    }
-
-    return ret;
-}
-
-// given path to image (wavelength).tif, return wavelength
-std::string ParseWavelength(fs::path imagePath)
+int main(int argc, char* argv[])
 {
-    auto name = imagePath.stem().string();
 
-    auto pos3 = name.find("-");
-    if (pos3 != std::string::npos) {
-        name.replace(pos3, 1, ".");
+    ///// Parse the cmd line /////
+    fs::path imgDir, outputPath;
+
+    // clang-format off
+    po::options_description options("Options");
+    options.add_options()
+        ("help,h","Show this message")
+        ("input-dir,i",po::value<std::string>()->required(),
+            "Input directory of wavelengths")
+        ("output-file,o", po::value<std::string>()->required(),
+            "Output file path");
+    // clang-format on
```

Comment:
add a `bit-depth` option, default to 16. Probably need to setup an enum so that the user can't just select any random bit-depth